lib/resources/geo-extracts.js

```diff
@@ -10,66 +10,51 @@
-};
-
 module.exports = (service, endpoint) => {
 
-  // Single-submission endpoints
-  service.get(`/projects/:projectId/forms/:xmlFormId/submissions/:instanceId.geojson`, endpoint.plain(async (container, context) =>
-    getSingleGeo(container, context, context.params.instanceId, null)
-  ));
-
-  service.get(`/projects/:projectId/forms/:xmlFormId/submissions/:rootId/versions/:instanceId.geojson`, endpoint.plain(async (container, context) =>
-    getSingleGeo(container, context, context.params.rootId, context.params.instanceId)
-  ));
-
-
-  // Bulk endpoints
-  service.get('/projects/:projectId/forms/:xmlFormId/submissions.geojson', endpoint.plain(async ({ Forms, GeoExtracts }, { auth, query, params }) => {
+  service.get('/projects/:projectId/forms/:xmlFormId/submissions.geojson', endpoint.plain(async ({ Forms, GeoExtracts, Actees }, { auth, query, params }) => {
 
     const form = await Forms.getByProjectAndXmlFormId(params.projectId, params.xmlFormId, Form.WithoutDef, Form.WithoutXml)
       .then(getOrNotFound)
       .then((foundForm) => auth.canOrReject('submission.list', foundForm));
 
-    return GeoExtracts.getSubmissionFeatureCollectionGeoJson(
+    const createResponse = () => GeoExtracts.getSubmissionFeatureCollectionGeoJson(
       form.id,
-      Sanitize.queryParamToArray(query.submissionID),
       Sanitize.queryParamToArray(query.fieldpath),
       Sanitize.queryParamToIntArray(query.submitterId, 'submitterId'),
       Sanitize.getTSTZRangeFromQueryParams(query),
       Sanitize.getSubmissionReviewStates(query.reviewState, 'reviewState'),
       isTrue(query.deleted),
       Number.parseInt(query.limit, 10) || null,
     ).then(json);
+
+    const acteeVersion = await Actees.getEventCount(form.acteeId);
+    // Weak etag, as the order in the resultset is undefined.
+    return withEtag(acteeVersion, createResponse, true);
   }));
 
-  service.get('/projects/:projectId/datasets/:datasetName/entities.geojson', endpoint.plain(async ({ Datasets, GeoExtracts }, { auth, query, params }) => {
+
+  service.get('/projects/:projectId/datasets/:datasetName/entities.geojson', endpoint.plain(async ({ Datasets, GeoExtracts, Actees }, { auth, query, params }) => {
 
     const foundDataset = await Datasets.get(params.projectId, params.datasetName, true)
       .then(getOrNotFound)
       .then((dataset) => auth.canOrReject('entity.list', dataset));
 
-    return GeoExtracts.getEntityFeatureCollectionGeoJson(
+    const createResponse = () => GeoExtracts.getEntityFeatureCollectionGeoJson(
       foundDataset.id,
-      Sanitize.queryParamToUuidArray(query.entityUUID, 'entityUUID'),
       Sanitize.queryParamToIntArray(query.creatorId, 'creatorId'),
       Sanitize.getTSTZRangeFromQueryParams(query),
       Sanitize.getEntityConflictStates(query.conflict, 'conflict'),
       isTrue(query.deleted),
       Number.parseInt(query.limit, 10) || null,
     ).then(json);
+
+    const acteeVersion = await Actees.getEventCount(foundDataset.acteeId);
```

Comment:
Should this count be done _before_ the response data is gathered to ensure the ETag is not more recent than the data it's served with?

Reply:
:+1: I get it now.  It might be more intuitive to move the `createResponse` definition after calculation of `acteeVersion`.